### ISSUE TITLE
Incoherent Result Labeling between GPTZero.sg and Github Code

### ISSUE THREAD

**Issue by slgphantom** [BurhanUlTayyab/DetectGPT#8]
When inputting the same text (generated by ChatGPT) into both GPTZero.sg's website and the GPT2PPL model on Github, the confidence levels were similar at around 50%, but the labeling results were different.

On GPTZero.sg, the label was 0, indicating that "this text is most likely generated by an A.I.", while using the code from the Github repository gave a label of 1, indicating that "this text is most likely written by a human".

Example input text: 
"Artificial Intelligence (AI) is the use of computers to perform tasks that would normally require human intelligence such as reasoning, perception, prediction, and planning. The ultimate goal of AI research is to create systems with human-like general intelligence, which remains a major challenge in the field. Researchers use a variety of methodologies and techniques such as heuristics, planning, mathematical simplification, and knowledge representation to achieve this goal. The current research focuses on using AI to recognize and respond to human emotions, rather than on creating AI systems that can experience emotions themselves. The backpropagation algorithm is a popular method used to train Artificial Neural Networks (ANNs) which is based on the principle of "backpropagation" of errors and it is found to be effective in training deep neural networks."

Additionally, it would be helpful if the threshold could be set as an input argument of the model so that users can customize it.

Thank you.

**Comment by slgphantom** [comment]
I tried the same text on GPTZero.sg with another machine, and the result aligns this time...

Any how, let me also share the result of using v1 (GPTZero) of the GPT2PPL model of the same text here:

```
Perplexity 10
Perplexity per line 14.0
Burstiness 20
(OrderedDict([('Perplexity', 10),
              ('Perplexity per line', 14.0),
              ('Burstiness', 20),
              ('label', 0)]),
 'The Text is generated by AI.')
```

**Comment by BurhanUlTayyab** [comment]
> Additionally, it would be helpful if the threshold could be set as an input argument of the model so that users can customize it.
Sure, we can add that as an input parameter for our next release.

**Comment by slgphantom** [comment]
Thank you for the prompt response!

![Screenshot_20230414-130040](https://user-images.githubusercontent.com/22796160/231945960-bb95dc13-221e-41ed-8a4c-8b1dd75e4321.png)

I noticed that the results are fluctuating. It might be helpful if you could test it on your device as well.

Could you please provide more information about the relationship between the confidence level, threshold, and labeling? As someone who is not well-versed in the internal calculations, I believe that the text is only labeled as "human" when the confidence level is above the threshold (0.7). However, this does not seem to be the case. It is frustrating to see that other methods are still reporting the text as being written by AI, such as GPTZero in v1 and DetectGPT on both your and the OG's website.

It would also be highly beneficial if the module could return an object that includes the confidence, label, and highlighted text (AI generated) so that users can better interpret the results and further apply them in their work.

**Comment by BurhanUlTayyab** [comment]
The results are same on every end. I think the fluctuation might occur because of different Pytorch / Transformer versions.
![image](https://user-images.githubusercontent.com/25680088/231948534-269c9bdb-9985-4181-89b9-386319ad8999.png)

For GPTZerov1, you can watch this video to understand how it works: https://www.youtube.com/watch?v=x9H-aY5sCDA (we've  added one more feature where perplexity per line is also calculated and aggregated).

For DetectGPT: We use Log Prob function from a LLM to detect text are similar to what it can produce. Also done line by line.
